docs/book/src/tasks/experimental-features/cluster-class/write-clusterclass.md

```diff
@@ -1,3 +1,570 @@
+different AWS regions. In this case, every cluster needs a Cluster-specific value for 
+`.spec.region` in `AWSCluster`. Use cases like this can be implemented with ClusterClass 
+patches.
+
+**Defining variables in the ClusterClass**
+
+The following example shows how variables can be defined in the ClusterClass.
+A variable definition specifies the name and the schema of a variable and if it is 
+required. The schema defines how a variable is defaulted and validated. It supports 
+a subset of the schema of CRDs. For more information please see the [godoc](https://doc.crds.dev/github.com/kubernetes-sigs/cluster-api/cluster.x-k8s.io/ClusterClass/v1beta1#spec-variables-schema-openAPIV3Schema).
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: ClusterClass
+metadata:
+  name: docker-clusterclass-v0.1.0
+spec:
+  ...
+  variables:
+  - name: imageRepository
+    required: true
+    schema:
+      openAPIV3Schema:
+        type: string
+        description: ImageRepository is the container registry to pull images from.
+        default: k8s.gcr.io
+        example: k8s.gcr.io
+```
+
+**Note**: The following basic types are supported: `string`, `integer`, `number` and `boolean`. Complex types 
+          are also supported (please see the corresponding section below).
+
+**Defining patches in the ClusterClass**
+
+The variable can then be used in a patch to set a field on a template referenced in the ClusterClass.
+The `selector` specifies on which template the patch should be applied. `jsonPatches` specifies which JSON 
+patches should be applied to that template. In this case we set the `imageRepository` field of the 
+`KubeadmControlPlaneTemplate` to the value of the variable `imageRepository`. For more information 
+please see the [godoc](https://doc.crds.dev/github.com/kubernetes-sigs/cluster-api/cluster.x-k8s.io/ClusterClass/v1beta1#spec-patches-definitions).
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: ClusterClass
+metadata:
+  name: docker-clusterclass-v0.1.0
+spec:
+  ...
+  patches:
+  - name: imageRepository
+    definitions:
+    - selector:
+        apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+        kind: KubeadmControlPlaneTemplate
+        matchResources:
+          controlPlane: true
+      jsonPatches:
+      - op: add
+        path: /spec/template/spec/kubeadmConfigSpec/clusterConfiguration/imageRepository
+        valueFrom:
+          variable: imageRepository
+```
+
+**Notes**:
+* Only fields below `/spec` can be patched.
+* Only `add`, `remove` and `replace` operations are supported.
+* It's only possible to append and prepend to arrays. Insertions at a specific index are 
+  not supported.
+* Be careful, appending or prepending an array variable to an array leads to a nested array
+  (for more details please see this [issue](https://github.com/kubernetes-sigs/cluster-api/issues/5944)).
+
+**Setting variable values in the Cluster**
+
+The user can now provide a value for the variable in the Cluster.
```

Comment:
```suggestion
After creating a ClusterClass with variable definition, the user will be allowed to provide a value for the variable in the Cluster as in the example below.
```

Reply:
Thx! I used your variant, but changed `will be allowed to provide` to `can now provide`